src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/AsnFormatter.OSX.cs

```diff
@@ -0,0 +1,220 @@
+                            needsConstructedFlag = true;
+                            break;
+                    }
+
+                    if (needsConstructedFlag &&
+                        (tag & DerSequenceReader.ConstructedFlag) == 0)
+                    {
+                        // All of the SEQUENCE types require the constructed bit,
+                        // or OpenSSL will have refused to print it.
+                        return null;
+                    }
+
+                    switch (nameType)
+                    {
+                        case GeneralNameType.OtherName:
+                            segments.Add("othername:<unsupported>");
+                            altNameReader.SkipValue();
+                            break;
+                        case GeneralNameType.Rfc822Name:
+                            segments.Add("email:");
+                            segments.Add(altNameReader.ReadIA5String());
+                            break;
+                        case GeneralNameType.DnsName:
+                            segments.Add("DNS:");
+                            segments.Add(altNameReader.ReadIA5String());
+                            break;
+                        case GeneralNameType.X400Address:
+                            segments.Add("X400Name:<unsupported>");
+                            altNameReader.SkipValue();
+                            break;
+                        case GeneralNameType.DirectoryName:
+                            // OpenSSL supports printing one of these, but the logic lives in X509Certificates,
+                            // and it isn't very common.  So we'll skip this one until someone asks for it.
+                            segments.Add("DirName:<unsupported>");
+                            altNameReader.SkipValue();
+                            break;
+                        case GeneralNameType.EdiPartyName:
+                            segments.Add("EdiPartyName:<unsupported>");
+                            altNameReader.SkipValue();
+                            break;
+                        case GeneralNameType.UniformResourceIdentifier:
+                            segments.Add("URI:");
+                            segments.Add(altNameReader.ReadIA5String());
+                            break;
+                        case GeneralNameType.IPAddress:
+                            segments.Add("IP Address");
+
+                            byte[] ipAddressBytes = altNameReader.ReadOctetString();
+
+                            if (ipAddressBytes.Length == 4)
+                            {
+                                // Add the colon and dotted-decimal representation of IPv4.
+                                segments.Add(
+                                    $":{ipAddressBytes[0]}.{ipAddressBytes[1]}.{ipAddressBytes[2]}.{ipAddressBytes[3]}");
+                            }
+                            else if (ipAddressBytes.Length == 16)
```

Comment:
I noticed the OpenSsl code is more forgiving here, just checking to see if a `:` is used (then assuming IPv6) and then allowing < 16 bytes under certain cases. Are we too restrictive here?